exporter/prometheusremotewriteexporter/exporter_test.go

```diff
@@ -1049,3 +1049,67 @@ func TestNoRetryOn4xx(t *testing.T) {
 	assert.True(t, consumererror.IsPermanent(err))
 	assert.Equal(t, 1, attempts)
 }
+
+// Simulate a timeout error that would result in an `http.client` returning an error.
+// This test retries once, and then on the second retry the server is configured to timeout.
+// This should lead to a permanent error.
+func TestNoRetryOnHttpClientError(t *testing.T) {
```

Comment:
are these tests running with the default timeout configurations? meaning that it will take a few seconds for it to complete? if yes, can we create a prw with a configuration that is more suitable for tests? (small/0 delay between retries)

Reply:
These tests are definitely retrying. When the unit tests are executed without the permanent error then the unit tests basically freezes as the time between timeouts is increased.

Reply:
> are these tests running with the default timeout configurations? meaning that it will take a few seconds for it to complete?
Not the first test. The timeout was changed to 10ms. On successful test run the client timeout will trigger at 10ms and if for some reason it does not then it is a 20ms status okay response